syncd/PortLinkEventDamper.h

```diff
@@ -0,0 +1,148 @@
+                    _In_ sai_object_id_t vid,
+                    _In_ sai_object_id_t rid,
+                    _In_ const sai_redis_link_event_damping_algo_aied_config_t &config);
+
+            virtual ~PortLinkEventDamper();
+
+            // Sets up the initial states for link event damping.
+            void setup();
+
+            // Handles and processes the port state change event.
+            void handlePortStateChange(
+                    _In_ sai_port_oper_status_t newPortState);
+
+            // Updates the link event damping configuration on the port.
+            void updateLinkEventDampingConfig(
+                    _In_ const sai_redis_link_event_damping_algo_aied_config_t &config);
+
+            // Handles for Selectable timer events.
+            void handleSelectableEvent();
+
+            struct DampingStats getDampingStats() const;
+
+            int getSelectableTimerFd();
+
+        private:
+            // Resets the timer.
+            void resetTimer(
+                    _In_ int64_t time_us);
+
+            // Cancels the timer by stopping the timer.
+            void cancelTimer();
+
+            // Advertises the port state to consumers.
+            void advertisePortState(
+                    _In_ sai_port_oper_status_t portState);
+
+            void updatePenaltyCeiling();
+
+            void updatePenalty();
+
+            bool isConfigSameAsRunningConfig(
+                    _In_ const sai_redis_link_event_damping_algo_aied_config_t &config);
+
+            // Checks if any config parameter value is 0.
+            bool configContainsZeroValueParam() const;
+
+            std::shared_ptr<NotificationHandler> m_notificationHandler;
+
+            // This is to add SelectableTimer.
+            std::shared_ptr<swss::Select> m_sel;
+
+            sai_object_id_t m_vid;
+            sai_object_id_t m_rid;
+
+            uint64_t m_maxSuppressTimeUsec;
+            uint64_t m_decayHalfLifeUsec;
+            uint32_t m_suppressThreshold;
+            uint32_t m_reuseThreshold;
+            uint32_t m_flapPenalty;
+            uint32_t m_accumulatedPenalty;
+            uint32_t m_penaltyCeiling;
+            bool m_dampingConfigEnabled;
+            bool m_dampingActive;
+            uint64_t m_lastPenaltyUpdateTimestamp;
+
+            // Timer to manage link event damping.
+            swss::SelectableTimer m_timer;
+
+            // Flag to annotate if timer is added to select.
+            bool m_timerAddedToSelect;
+            bool m_timerActive;
+
+            // Advertised port's oper status to consumers.
+            sai_port_oper_status_t m_advertisedState;
+
+            // Actual port's oper status in the ASIC.
+            sai_port_oper_status_t m_actualState;
+
+            // Damping related statistics.
+            struct DampingStats m_stats;
+
+            friend class PortLinkEventDamperPeer;
+            friend class LinkEventDamperPeer;
```

Comment:
in entire project we don't use friend class, please explain in details why this is exactly what you need and it cant be done in other way?